src/contexts/productsDetail-context.js

```diff
@@ -0,0 +1,24 @@
+import React, { useState, useContext, useEffect, createContext } from "react";
+import axios from "axios";
+
+const productsDetailContext = createContext({ products: []})
+
+const ProductsDetailProvider = ({ children }) => {
+    
+    const [products, setProducts] = useState([]);
+
+    useEffect(() =>
+        (async () => {
+            const ProductResponse = await axios.get("api/products")
+            setProducts(ProductResponse.data.products)
+        })(), []
+    );
```

Comment:
you can destructure productResponse here like {productResponse}

Reply:
okay, will try doing it the suggested way, thanks for the suggestion!!